### ISSUE TITLE
Use actual probabilities in het pulldown

### ISSUE THREAD

**Issue by davidbenjamin** [broadinstitute/gatk-protected#67]
The current approach is to perform a p-value test under a null hypothesis that a site is het to decide whether to include it.  The problem with this is the problem of all frequentist hypothesis testing in that it is totally agnostic of the other hypothesis, which in this case we know is that the site is homozygous.  This is a fairly simple application of Bayes' Theorem and will avoid throwing out perfectly good het sites. It could in principle also avoid treating homozygous sites as hets, but this possibility is unlikely due to a filter for low-coverage sites.  Still, by using a probabilistic approach one can dispense with this ad hoc filter, which will in turn let us keep more het sites.

**Comment by davidbenjamin** [comment]
One other danger in the current approach is that if some true het site has large allelic bias, it could have alt/ref counts that are skewed enough to give a very small p value for heterozygosity assuming no bias or small bias.  For example, alt/ref counts of 60/140 are obviously a het, even though their p-value under the null hypothesis of binomial counts is small.

**Comment by LeeTL1220** [comment]
Does this happen in practice (in a normal)?  Enough that we care?

**Comment by davidbenjamin** [comment]
There's a decent chance that we don't care. I'll do some back-of-the-envelope calculations to see under what conditions the p-value method loses hets.

**Comment by davidbenjamin** [comment]
I did some simple calculations, and it's worth it to do something probabilistic.  However, the stuff I have in the proposed methods docs is overkill (what was I thinking???).  We can do a lot better than the p-value while keeping things simple.  Here are some highlights:

*By definition, a p-value of 0.05 means that we are guaranteed to throw out 5% of all true hets.
*But it gets worse if there is even quite moderate allelic bias.  For example, if the total depth is 50 reads, 19 alt reads is the threshold at which p = 0.05.  But with a modest allelic bias of 47% alt probability this many alt reads or fewer occurs 13% of the time, so we are quite likely to throw away this het.
*If there are 10 total reads and 2 alt reads, p = 0.054 and we accept it as a het, when it could easily be a hom ref with sequencing errors.
*Note that all of these calls are obvious to us without any sort of statistical model.

Honestly, something crazily simple like a hard threshold of 3 minor reads would probably do quite well.  However, almost as simple but better founded would be to posit some reasonable guess for the error rate, say 1% (the result won't be very sensitive to this) and no allelic bias.  We can filter by posterior probability of a het, which maps one-to-one to the likelihood ratio of het to hom.  This ratio is, if we have n alt/ref refs and m ref/alt reads, where WLOG n < m: (1/2)^(n+m)/( e^m(1-e)^n).

In conclusion, for a code change that will take an hour or two and won't increase complexity we get 5% more hets for free.

**Comment by LeeTL1220** [comment]
Delayed by fargo (possibly)

**Comment by samuelklee** [comment]
Moving this to beta (see discussion in #314).

**Comment by davidbenjamin** [comment]
@mbabadi You address this in PR #444.
